provider/providers.go

```diff
@@ -147,3 +148,25 @@ func NewTwitter(p Params) Oauth1Handler {
 		},
 	})
 }
+
+// NewBattlenet makes Battle.net oauth2 provider
+func NewBattlenet(p Params) Oauth2Handler {
+	return initOauth2Handler(p, Oauth2Handler{
+		name: "battlenet",
+		endpoint: oauth2.Endpoint{
+			AuthURL:   "https://eu.battle.net/oauth/authorize",
+			TokenURL:  "https://eu.battle.net/oauth/token",
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+		scopes:  []string{},
```

Comment:
does it support scopes? if so, we should set a minimal one sufficient for the basic info we get